music_this_week_app/templates/music_this_week/setup.html

```diff
@@ -12,23 +12,75 @@
+            <p><input id = "hidden_lat" hidden = true name = "hidden_lat"></p>
+            <p><input id = "hidden_lon" hidden = true name = "hidden_lon"></p>
             <input hidden=true name="nResults" value="200">
             <button type="submit" id="search-button" onclick="buttonClick(this.form)">Search</button>
         </form>
         <p>&nbsp;</p>
         <p id="wait-text" hidden="True">Please be patient. In few minutes a playlist called musicThisWeek will appear in your spotify app.<br><br>If you close this page, the playlist will still be created in the background.</p>
     </div>
-
     <script>
+        //init a variable that will be true if checkbox clicked and user approves location request
+        var userLocationProvided = false;
         //When Search button is pressed, change the UI and then submit the request.
         function buttonClick(form) {
             document.getElementById('search-button').innerHTML = "Searching...";
             document.getElementById('search-button').disabled = "True";
             document.getElementById('wait-text').hidden = false;
             form.submit();
         }
+        //When checkbox is clicked request users location if we havent already
+        function clickCheckBox(){
+            console.log("box checked");
+            console.log(userLocationProvided);
+            var checkbox = document.getElementById('latlng_checkbox');
+            if ((!userLocationProvided)  && (checkbox.checked)){
+                //uncheck box until location obtained, prevents errors from user double click
+                checkbox.checked = false;
+                getUserLocation();
+                }
+            else{
+                toggleLocationField();
+                }            
+
+        }
+        //Hides and Shows location field of form
+        function toggleLocationField(){
+            var loc_field = document.getElementById('loc_field');
+            if (loc_field.hidden){
+                loc_field.hidden = false;
+                }
+            else{
+                loc_field.hidden = true;
+                }
+        }
+
+        function locationProvided(position) {
+            loc_field.hidden = true;
+            var lat = position.coords.latitude.toString();
+            var lon = position.coords.longitude.toString();
+            document.getElementById('hidden_lat').value = lat;
+            document.getElementById('hidden_lon').value = lon;
+            document.getElementById('latlng_checkbox').checked = true;
+            //update global bool so location not requested after user accepts
+            userLocationProvided = true; 
+	}
+ 
+        function locationError(error){
+            document.getElementById('latlng_checkbox').checked = false;
+            document.getElementById('location_options').innerHTML = '<strong>Location:</strong> <input id = "loc_field" type="text" name="location" value="{{location}}">' ;
```

Comment:
It'd be best to avoid defining this element's inner HTML in two places in the same file. If it ever has to change, you'd have to know to change it twice.

Any way to avoid this?